src/arrayMethodJoin.js

```diff
@@ -5,8 +5,26 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let newSeparator = separator;
+
+    if (separator === undefined) {
```

Comment:
You can use ternary operator to make this if unnecessary.

```js
const newSeparator = separator === undefined ? ',' : separator;
```

This would assign separator if it exists, if it doesn't

Reply:
Good idea